src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -171,6 +174,114 @@ private byte[] GetBytesFromSlave(int count, TimeInterval timeInterval)
             return result;
         }
 
+        private void PollForError(TimeInterval timeInterval){
+            var errors = ReadCommonErrorStatusBytes(timeInterval);
+            var sw = new Stopwatch();
+            sw.Start();
+            while (errors[0] == 0x00 && sw.Elapsed < new TimeSpan(0,0,3)){
+                errors = ReadCommonErrorStatusBytes(timeInterval);
+            }
+        }
+
+        // When we need to read for more than 255 bytes, we will need this method
+        private byte[] GetLongBytes(int count, TimeInterval timeInterval) {            
+            var result = new byte[count];
+            int rounds = count /255;
+            int remainder = count % 255; 
+            for (int i = 0; i <= rounds; i ++) {
+                int toRead = i == rounds ? remainder : 255;
+                Array.Copy(GetBytesFromSlave(toRead, timeInterval), 0, result, i * 255, toRead);
+            }
+            return result;
+        }
+
+        public void StartSocketServer(TimeInterval timeInterval)
+        {  
+            Console.WriteLine(currentSlave.GetType().Name);
+            string path = "/tmp/i2c.sock";
+            if (File.Exists(path))
+            {
+                File.Delete(path);
+            }
+            var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+            socket.Bind(new UnixEndPoint(path));
+            socket.Listen(0);
+            Console.WriteLine("Server started waiting for client to connect...");
+            Console.WriteLine("Starting on: "+ path);
+            currentSlave.Read(0);
+            var connection = socket.Accept();
+            Console.WriteLine("Connection established");
+
+            var commandBuffer = new byte[1];
+            int length;
+            while(SocketConnected(connection)){
+                try {
+                    PollForSocketData(connection, (int)numOfBytes.MessageType, commandBuffer);
+                }
+                catch(Exception e)
+                {
+                    Console.WriteLine("{0} Exception caught.", e);
```

Comment:
What's the idea behind `continue` here? Is there any type of error that we could recover from?

Reply:
It can continue receiving bytes even after the error. But it's probably not ideal, I was doing some tests around at the very beginning stage. I'll change it now.